### PR TITLE
feat(vpc): doc for managed routes public beta

### PR DESCRIPTION
**Pull request by RoRoJ**
Document managed routes for VPCs in public beta

image

image

### CODE REVIEW

File: network/vpc/how-to/manage-routing.mdx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+## How to activate routing on a VPC
+
+You must create a new VPC in order to activate routing. It is not yet possible to activate routing on pre-existing VPCs.
+
+1. Click **VPC** in the **Network** section of the side menu. The list of your VPCs displays.
+
+2. Click **+ Create VPC**. The VPC creation wizard displays.
+
+    <Lightbox src="TODO SCREENSHOT THAT INCLUDES CONFIGURE OPTIONS" alt="" />
```
Comment (nerda-codes) [inline network/vpc/how-to/manage-routing.mdx:42]: Reminder for the screenshot 😉

File: network/vpc/how-to/manage-routing.mdx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+3. Choose a **region** for your VPC, and enter a **name** and (optionally) **tags**.
```
Comment (nerda-codes) [inline network/vpc/how-to/manage-routing.mdx:44]:
```suggestion
3. Choose a **region** for your VPC, and enter a **name** and optional **tags**.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+1. Click **VPC** in the **Network** section of the side menu. The list of your VPCs displays.
+
+2. Click the VPC on which you have activated routing.
+
+3. Click the **Routing** tab.
+
+    If you have not yet created any Private Networks in the VPC, the route table will be empty. Routes are automatically generated and added to the route table when you either:
+    
+    - Create a Private Network in the VPC (this generates a **local subnet route**, which allows the VPC to automatically route traffic between Private Networks), or 
+    - Attach a Public Gateway to a Private Network and set it to advertise a default route. This generates a **default route to the internet**.
```
Comment (nerda-codes) [inline network/vpc/how-to/manage-routing.mdx:69]:
```suggestion
    - Attach a Public Gateway to a Private Network and set it to advertize a default route. This generates a **default route to the internet**.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    
+    When your route table starts to populate, it will look something like this:
+
+    <Lightbox src="TODO SCREENSHOT OF ROUTE TABLE" alt="" />
```
Comment (nerda-codes) [inline network/vpc/how-to/manage-routing.mdx:73]: Same as above

File: network/vpc/how-to/manage-routing.mdx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    For help with understanding the route table and how to read it, [read more about route tables](/network/vpc/concepts#route-table). Note that it is not currently possible to create your own custom routes, but this feature will be coming soon.
```
Comment (nerda-codes) [inline network/vpc/how-to/manage-routing.mdx:75]:
```suggestion
    For help with understanding the route table and how to read it, [refer to our documentation about route tables](/network/vpc/concepts#route-table). Note that it is not currently possible to create your own custom routes, but this feature will be coming soon.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+    As all routes are currently managed by Scaleway, you do not need to take any action with your route table. If and when you delete Private Networks from the VPC, or delete a Public Gateway advertising the default route, these routes will be automatically deleted from your route table.
```
Comment (nerda-codes) [inline network/vpc/how-to/manage-routing.mdx:77]:
```suggestion
    As all routes are currently managed by Scaleway, you do not need to take any action with your route table. If and when you delete Private Networks from the VPC, or delete a Public Gateway advertizing the default route, these routes will be automatically deleted from your route table.
```

File: network/vpc/concepts.mdx [hunk 7]
```diff
@@ -83,6 +83,44 @@ While DHCP is built into all new Private Networks, it may not be automatically a
...
 
+## Route table
+
+<Message type="note">
+This feature is currently in public beta, and only available when you create a new VPC.
+</Message>
+
+A route table is used to manage and control the routing of traffic within a VPC. The routes within a route table tell the VPC where to send traffic trying to get to a specific destination IP address. One line in the route table corresponds to one route. A route consists of:
+
+- A **destination** IP or IP range. This specifies that the route applies to traffic with a matching destination IP.
```
Comment (bene2k1) [inline network/vpc/concepts.mdx:94]:
```suggestion
- A **destination** IP or IP range. This specifies that the route applies to traffic with a matching destination IP or range.
```

**Review by bene2k1 — APPROVED** [review]

File: network/vpc/concepts.mdx [hunk 8]
```diff
@@ -83,6 +83,44 @@ While DHCP is built into all new Private Networks, it may not be automatically a
...
+
+<Message type="note">
+This feature is currently in public beta, and only available when you create a new VPC.
+</Message>
+
+Routing allows Private Networks in the same VPC to communicate with each other, via automatically-created and managed routes.
+
+Every time you create a Private Network in a VPC in which you have activated routing, a managed route is automatically created and added to the VPC's [route table](#route-table). This route allows the VPC to automatically route packets to resources attached to that Private Network, even if they originate from a resource attached to a different Private Network on the VPC.
+
+Managed routes are also automatically added to the VPC's routing table when you attach a Public Gateway to a Private Network, and tell it to [advertize the default route](/network/public-gateways/concepts/#default-route). This type of managed route allows traffic to be forwarded to addresses outside the VPC (i.e. the public internet) via the Public Gateway.
```
Comment (jcirinosclwy) [inline network/vpc/concepts.mdx:114]:
```suggestion
Managed routes are also automatically added to the VPC's routing table when you attach a Public Gateway to a Private Network, and tell it to [advertise the default route](/network/public-gateways/concepts/#default-route). This type of managed route allows traffic to be forwarded to addresses outside the VPC (i.e. the public internet) via the Public Gateway. 
```

File: network/vpc/concepts.mdx [hunk 9]
```diff
@@ -83,6 +83,44 @@ While DHCP is built into all new Private Networks, it may not be automatically a
...
+
+Routing allows Private Networks in the same VPC to communicate with each other, via automatically-created and managed routes.
+
+Every time you create a Private Network in a VPC in which you have activated routing, a managed route is automatically created and added to the VPC's [route table](#route-table). This route allows the VPC to automatically route packets to resources attached to that Private Network, even if they originate from a resource attached to a different Private Network on the VPC.
+
+Managed routes are also automatically added to the VPC's routing table when you attach a Public Gateway to a Private Network, and tell it to [advertize the default route](/network/public-gateways/concepts/#default-route). This type of managed route allows traffic to be forwarded to addresses outside the VPC (i.e. the public internet) via the Public Gateway. 
+
+<Message type="note">
+Public Gateways remain scoped to the Private Network to which they are attached. They do not advertize the default route on other Private Networks in the VPC. For example, an Instance attached to Private Network A will not be able to access the internet via a Public Gateway in Private Network B.
```
Comment (jcirinosclwy) [inline network/vpc/concepts.mdx:117]:
```suggestion
Public Gateways remain scoped to the Private Network to which they are attached. They do not advertise the default route on other Private Networks in the VPC. For example, an Instance attached to Private Network A will not be able to access the internet via a Public Gateway in Private Network B.
```

File: network/vpc/concepts.mdx [hunk 10]
```diff
@@ -83,6 +83,44 @@ While DHCP is built into all new Private Networks, it may not be automatically a
...
+</Message>
+
+Note that routing cannot be deactivated after creation of the VPC. More routing features are planned for the future, including custom routes, routing on existing VPCs and more.
```
Comment (jcirinosclwy) [inline network/vpc/concepts.mdx:120]:
```suggestion
Note that routing cannot be deactivated after the creation of the VPC. More routing features are planned for the future, including custom routes, routing on existing VPCs, and more.
```

File: network/vpc/how-to/attach-resources-to-pn.mdx [hunk 11]
```diff
@@ -137,5 +137,5 @@ You can also detach a resource to a Private Network from the resource's own sect
 
 <Navigation title="See also">
   <PreviousButton to="/network/vpc/how-to/activate-dhcp/">How to activate DHCP on a legacy Private Network</PreviousButton>
-  <NextButton to="/network/vpc/how-to/delete-private-network/">How to delete a Private Network</NextButton>
-</Navigation>
\ No newline at end of file
+  <NextButton to="/network/vpc/how-to/manage-routing/">How to manage routing on a VPC</NextButton>
```
Comment (jcirinosclwy) [inline network/vpc/how-to/attach-resources-to-pn.mdx:140]: Considering the PR from Aymar, do we still keep the See also block here?

File: network/vpc/how-to/manage-routing.mdx [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+tags: private-network vpc routing route-table routes default-route local-route subnet
+dates:
+  validation: 2024-04-09
+  posted: 2024-04-09
+categories:
+  - network
+---
+
+<Message type="important">
+This document refers to the **routing** feature which is currently in public beta. Routing can only be activated when you create a new VPC. It cannot yet be activated on pre-existing VPCs.
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:17]:
```suggestion
This document refers to the **routing** feature, which is currently in public beta. Routing can only be activated when you create a new VPC. It cannot yet be activated on pre-existing VPCs.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+Currently, only auto-created routes are available, which are fully managed by Scaleway. Two types of managed route exist:
+- **Local subnet route**: Generated when you create a Private Network in a VPC. Allows traffic to be routed between different Private Networks in the VPC.
+- **Default route to internet**: Generated when you attach a Public Gateway to a Private Network in the VPC, and set it to advertise a [default route](/network/public-gateways/concepts/#default-route). Allows traffic to be routed to addresses outside the VPC (i.e. the public internet) via the gateway. 
+
+<Message type="note">
+Public Gateways remain scoped to the Private Network to which they are attached. They do not advertize the default route on other Private Networks in the VPC. For example, an Instance attached to Private Network A will not be able to access the internet via a Public Gateway in Private Network B.
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:27]:
```suggestion
Public Gateways remain scoped to the Private Network to which they are attached. They do not advertise the default route on other Private Networks in the VPC. For example, an Instance attached to Private Network A will not be able to access the internet via a Public Gateway in Private Network B.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+2. Click **+ Create VPC**. The VPC creation wizard displays.
+
+    <Lightbox src="scaleway-create-vpc-routing.webp" alt="" />
+
+3. Choose a **region** for your VPC, and enter a **name** and optional **tags**.
+
+4. Under **Configure options**, slide the toggle to activate routing.
+
+    <Message type="important">
+    Routing cannot be deactivated after creation of the VPC.
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:51]:
```suggestion
    Routing cannot be deactivated after the creation of the VPC. 
```

File: network/vpc/how-to/manage-routing.mdx [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+1. Click **VPC** in the **Network** section of the side menu. The list of your VPCs displays.
+
+2. Click the VPC on which you have activated routing.
+
+3. Click the **Routing** tab.
+
+    If you have not yet created any Private Networks in the VPC, the route table will be empty. Routes are automatically generated and added to the route table when you either:
+    
+    - Create a Private Network in the VPC (this generates a **local subnet route**, which allows the VPC to automatically route traffic between Private Networks), or 
+    - Attach a Public Gateway to a Private Network and set it to advertize a default route. This generates a **default route to the internet**.
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:71]:
```suggestion
    - Attach a Public Gateway to a Private Network and set it to advertise a default route. This generates a **default route to the internet**.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+    
+    When your route table starts to populate, it will look something like this:
+
+    <Lightbox src="scaleway-route-table.webp" alt="" />
+
+    For help with understanding the route table and how to read it, [refer to our documentation about route tables](/network/vpc/concepts#route-table). Note that it is not currently possible to create your own custom routes, but this feature will be coming soon.
+
+    As all routes are currently managed by Scaleway, you do not need to take any action with your route table. If and when you delete Private Networks from the VPC, or delete a Public Gateway advertizing the default route, these routes will be automatically deleted from your route table.
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:79]:
```suggestion
    As all routes are currently managed by Scaleway, you do not need to take any action with your route table. If and when you delete Private Networks from the VPC, or delete a Public Gateway advertising the default route, these routes will be automatically deleted from your route table.
```

File: network/vpc/how-to/manage-routing.mdx [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+
+### How to view VPC routes in IPV6 
+
+Scaleway VPC routing supports both IPv4 and IPv6 protocols. Managed routes to Private Networks are simultaneously generated for both IPV4 and IPV6, and both are added to the route table. Use the toggle above the route table to switch from the default view of **IPV4** routes, to a view of **IPV6** routes.
+
+## Routing limitations
+
+The following limitations apply to VPC routing during this public beta phase:
+
+- Managed Databases are not currently compatible with routing. The VPC cannot automatically route between Managed Databases on different Private Networks, or (for example) between a Managed Databases on one Private Network and an Instance on a different Private Network.
```
Comment (jcirinosclwy) [inline network/vpc/how-to/manage-routing.mdx:89]:
```suggestion
- Managed Databases are not currently compatible with routing. The VPC cannot automatically route between Managed Databases on different Private Networks, or (for example) between a Managed Database on one Private Network and an Instance on a different Private Network.
```

**Review by jcirinosclwy — APPROVED** [review]

File: network/vpc/how-to/attach-resources-to-pn.mdx (hunk 11 above)
Comment (RoRoJ) [reply network/vpc/how-to/attach-resources-to-pn.mdx:140]: I think we do until the new system is put in place

**Pull request merged**